sphinx_search/_static/js/rtd_sphinx_search.js

```diff
@@ -71,97 +75,248 @@ const createDomNode = (nodeName, attributes) => {
+    if (resultData.highlight !== undefined && resultData.highlight !== null) {
+        if (
+            resultData.highlight.title !== undefined &&
+            resultData.highlight.title !== null
+        ) {
+            title.innerHTML = resultData.highlight.title;
+        } else {
+            title.innerHTML = resultData.title;
+        }
+    } else {
+        title.innerHTML = resultData.title;
+    }
+
+    page_link.appendChild(title);
+    content.appendChild(page_link);
+    content.appendChild(createDomNode("br"));
+
+    // all the sections of the page.
+    let sections = resultData.inner_hits.sections;
+
+    // all the sphinx domains of the page.
+    let domains = resultData.inner_hits.domains;
+
+    if (sections !== undefined && sections !== null) {
+        for (let i = 0; i < sections.length; ++i) {
+            COUNT += 1;
+            let section_link = createDomNode("a", {
+                href: `${page_link.href}#${sections[i]._source.id}`
+            });
+            let section_title = createDomNode("span", {
+                class: "search__result__subheading"
+            });
+
+            // if section title is present in the highlighted field, use that.
+            // else use the title present in the _souce.
+            if (
+                sections[i].highlight["sections.title"] !== undefined &&
+                sections[i].highlight["sections.title"].length >= 1
+            ) {
+                section_title.innerHTML =
+                    sections[i].highlight["sections.title"][0];
+            } else {
+                section_title.innerHTML = sections[i]._source.title;
+            }
+
+            let section_content = createDomNode("p", {
+                class: "search__result__content"
+            });
+
+            // if section content is present in the highlighted field, use that.
+            // else use the content present in the _souce.
+            if (
+                sections[i].highlight["sections.content"] !== undefined &&
+                sections[i].highlight["sections.content"].length >= 1
+            ) {
+                section_content.innerHTML =
+                    "... " +
+                    sections[i].highlight["sections.content"][0] +
+                    " ...";
+            } else {
+                section_content.innerHTML =
+                    sections[i]._source.content.substring(0, 50) + " ...";
+            }
+
+            let outer_div = createDomNode("div", {
+                class: "outer_div_page_results",
+                id: "hit__" + COUNT
+            });
+            outer_div.appendChild(section_title);
+            outer_div.appendChild(section_content);
+            section_link.appendChild(outer_div);
+            content.appendChild(section_link);
+            content.appendChild(createDomNode("br", { class: "br-for-hits" }));
+        }
+    }
+
+    if (domains !== undefined && domains !== null) {
```

Comment:
Are we always showing sections before domains in this logic?

Reply:
Yes.
It is because we are getting the results of sections and domains separately.
sections results are sorted and domains results are sorted -- but there is no sorting present which invloves both.

Reply:
I believe ES should return the score for each object, and we're removing it in the web API. We could probably be  passing along the score to the JS, so that we could use it more smartly for sorting.

Reply:
Yes. Score is present in each result.
Where should the sorting be done? server side or client side?

Reply:
I think a server side sort is reasonable so that the search results on readthedocs.org and in-doc search mostly match.

Reply:
I have implemented a server side sort. But I'm still concerned about its efficiency as I have not implemented any algorithm.